include/random_walks/uniform_accelerated_billiard_walk.hpp

```diff
@@ -127,30 +239,52 @@ struct AcceleratedBilliardWalk
                 }
 
                 _lambda_prev = dl * pbpair.first;
-                _p += (_lambda_prev * _v);
+                if constexpr (std::is_same<MT, Eigen::SparseMatrix<NT, Eigen::RowMajor>>::value) {
+                    _update_parameters.moved_dist = _lambda_prev;
+                    typename Point::Coeff b = P.get_vec();
+                    NT* b_data = b.data();
```

Comment:
since be the vector b doesn't change, why don't you initialize the `b_data` before the loop?